### PR TITLE
Bvego authentication

### PR DESCRIPTION
**Pull request by bvego**
Implemented very basic authentication and styles for the authentication components.

### CODE REVIEW

File: webpack.config.dev.js [hunk 1]
```diff
@@ -21,8 +21,8 @@ module.exports = {
         extensions: ['.js', '.jsx', '.scss'],
         alias: {
```
Comment (ikovic) [inline webpack.config.dev.js]: Aliases should always be set the same both for `dev` and `prod` environments. Maybe extracting that to a common module could prevent some nasty bugs from happening?

File: src/store.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import { createStore } from 'redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
 import rootReducer from './reducers';
 
-const store = createStore(rootReducer);
+const store = createStore(rootReducer, applyMiddleware(thunk));
```
Comment (ikovic) [inline src/store.js]: Add this too: https://www.npmjs.com/package/redux-devtools-extension
Install the extension for your browser, that should save a lot of debugging time.

File: src/routes/Authentication/index.jsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 import { Route } from 'react-router-dom';
 import Login from 'routes/Authentication/containers/Login';
+import Register from 'routes/Authentication/containers/Register';
 
 const Authentication = () => (
-    <div>
-        test
-    </div>
+    <Fragment>
+        <Route exact path="/" component={Login} key="login-route" />
```
Comment (ikovic) [inline src/routes/Authentication/index.jsx]: No need for `key` attribute here, it's useful only for rendering dynamic lists.

File: src/routes/Authentication/containers/Register/index.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import propTypes from 'prop-types';
+import { registerUser } from 'reducers/user';
+import Hero from 'routes/Authentication/Components/Hero';
+import './Register.scss';
+
+class Register extends Component {
+    constructor(props) {
```
Comment (ikovic) [inline src/routes/Authentication/containers/Register/index.jsx]: No need to pass `props` to constructor if you are not going to seed the initial state with it.

File: src/routes/Authentication/containers/Register/index.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        super(props);
+        this.state = {};
+        this.onSubmit = this.onSubmit.bind(this);
```
Comment (ikovic) [inline src/routes/Authentication/containers/Register/index.jsx]: No need to use bind here. Just write instance methods with arrow functions:
```
onSubmit = event => ...
```
Also instance properties can be set in a shorter way:
```
state = {}
```

File: src/routes/Authentication/containers/Register/index.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                            placeholder="Password"
+                            className="input"
+                            onChange={event => this.setState({ password: event.target.value })}
+                        />
+                    </div>
+                    <div className="field is-grouped is-grouped-centered">
+                        <span className="control">
+                            <button
+                                type="submit"
+                                className={`button is-primary ${
```
Comment (ikovic) [inline src/routes/Authentication/containers/Register/index.jsx]: Perfect use case for [classnames](https://www.npmjs.com/package/classnames)

File: src/routes/Authentication/containers/Login/index.jsx [hunk 7]
```diff
@@ -1,15 +1,89 @@
...
+                            placeholder="Password"
+                            className="input"
+                            onChange={event => this.setState({ password: event.target.value })}
+                        />
+                    </div>
+                    <div className="field is-grouped is-grouped-centered">
+                        <span className="control">
+                            <button
+                                type="submit"
+                                className={`button is-primary ${
```
Comment (ikovic) [inline src/routes/Authentication/containers/Login/index.jsx]: Check the `classnames` comment below

File: src/routes/Authentication/containers/Login/index.jsx [hunk 8]
```diff
@@ -1,15 +1,89 @@
...
+import propTypes from 'prop-types';
+import { loginUser } from 'reducers/user';
+import Hero from 'routes/Authentication/Components/Hero';
+import './Login.scss';
 
 class Login extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+        this.onSubmit = this.onSubmit.bind(this);
```
Comment (ikovic) [inline src/routes/Authentication/containers/Login/index.jsx]: Same as constructor and `bind` comment below.

File: src/reducers/user.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+import * as UC from 'constants/user';
```
Comment (ikovic) [inline src/reducers/user.js]: Action types and action creators should be in the same module following the `ducks` structure.
Constants should be used for things like API URLs, magic string, etc...

File: src/constants/user.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export const LOGIN_START = 'LOGIN_START';
```
Comment (ikovic) [inline src/constants/user.js]: Pls read the comment below regarding the location of action types.
Other than that, this way of naming action types is a bit unsafe. 
Even though you are namespacing the values when importing them, there is no guarantee that the actual strings are unique. It's possible you'll have a naming collision. Good way to mitigate this risk is following the `ducks` structure once again - all action types should be created by this formula: `appName/reducerName/ACTION_NAME`.

File: package.json [hunk 11]
```diff
@@ -38,6 +42,7 @@
     "file-loader": "^1.1.6",
     "happypack": "^4.0.0",
     "html-webpack-plugin": "^2.30.1",
+    "json-server": "^0.12.1",
     "node-sass": "^4.7.2",
     "prettier": "1.10.2",
     "react-hot-loader": "^4.0.0-beta.14",
```
Comment (ikovic) [inline package.json]: Maybe add a script for running the backend server here too?

File: .prettierrc [hunk 12]
```diff
@@ -3,5 +3,6 @@
     "useTabs": false,
     "printWidth": 100,
     "tabWidth": 4,
```
Comment (ikovic) [inline .prettierrc]: Not much point in talking about spacing, but maybe try with `tabWidth` of 2 and `printWidth` of 120?
We are all using high res screens so more characters could fit in a single line. Making spaces smaller could help with readability.

File: webpack.config.dev.js (hunk 1 above)
Comment (bvego) [reply webpack.config.dev.js]: Yip. Fixed in cbd0a97

File: src/store.js (hunk 2 above)
Comment (bvego) [reply src/store.js]: Added in e0a3efd

File: src/routes/Authentication/index.jsx (hunk 3 above)
Comment (bvego) [reply src/routes/Authentication/index.jsx]: At first I was returning the routes as an array, this was a leftover that I forgot to remove
Removed in 989b00e

File: src/routes/Authentication/containers/Register/index.jsx (hunk 4 above)
Comment (bvego) [reply src/routes/Authentication/containers/Register/index.jsx]: Added in 3d26ec2

File: src/routes/Authentication/containers/Register/index.jsx (hunk 5 above)
Comment (bvego) [reply src/routes/Authentication/containers/Register/index.jsx]: Added in 3d26ec2

File: src/routes/Authentication/containers/Register/index.jsx (hunk 6 above)
Comment (bvego) [reply src/routes/Authentication/containers/Register/index.jsx]: I never used the classnames package, not sure if I implemented it the right way (by this i mean whether I should first initialize a variable and use it in className={variable name} or set it directly as I did in the commit)

Added in f269efb

File: src/routes/Authentication/containers/Login/index.jsx (hunk 7 above)
Comment (bvego) [reply src/routes/Authentication/containers/Login/index.jsx]: Added in f269efb

File: src/routes/Authentication/containers/Login/index.jsx (hunk 8 above)
Comment (bvego) [reply src/routes/Authentication/containers/Login/index.jsx]: Added in 3d26ec2

File: src/reducers/user.js (hunk 9 above)
Comment (bvego) [reply src/reducers/user.js]: I've misunderstood the 'contants' term. I always placed API urls and similar stuff in a 'config' folder, and had a separate file for constants related to redux

Hopefully implemented correctly in 55c162f

File: src/constants/user.js (hunk 10 above)
Comment (bvego) [reply src/constants/user.js]: Changed in 55c162f

File: package.json (hunk 11 above)
Comment (bvego) [reply package.json]: Added in 7cd0b7c

**Pull request merged**